core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1006,7 +1006,7 @@ class DAGScheduler(
         runningStages -= stage
         return
     }
-
+    val tokens = KerberosUtil.getHadoopDelegationTokens
```

Comment:
See my comment about below about how current spark (on yarn) handles security https://github.com/apache/spark/pull/16788/files#r105319503.

@tgravescs or @vanzin can correct me if I am wrong (in case I am misremembering) - 
In a secure hdfs, it is not necessary for principal/keytab to be provided - if the job will finishes before token renewal is necessitated.

Given above : The call chain in KerberosUtil.getHadoopDelegationTokens will throw an exception if they are missing if ugi security is enabled.
I am not sure if this is a requirement in mesos, but it is not for yarn.